helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -76,21 +76,81 @@ receivers:
       {{- end }}
 
       # Receivers for collecting k8s control plane metrics.
+      # Verified with Kubernetes v1.22 and Openshift v4.9.
+      # Below, the TLS certificate verification is often skipped because the k8s default certificate is self signed and
+      # will fail the verification.
       {{- if .Values.autodetect.controlPlane }}
+      smartagent/coredns:
+        {{- if eq .Values.distribution "openshift" }}
+        rule: type == "pod" && namespace == "openshift-dns" && name contains "dns"
+        {{- else }}
+        rule: type == "pod" && labels["k8s-app"] == "kube-dns"
+        {{- end }}
+        config:
+          extraDimensions:
+            metric_source: k8s-coredns
```

Comment:
I didn't catch this on the previous PR for other control plan receivers - I think it's OK to keep these for now, but probably in the future when we switch these to using the otel prometheus scraper we should look into modifying content instead of adding extra dimensions. If the metric names are unique enough anyway, they won't be needed.

Reply:
Good to know. I'll keep this in mind. These added dimensions are currently built into our k8s services dashboard, we can change this in the future.